Comp2024/src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -48,19 +65,62 @@ public Shooter( )
         && PhoenixUtil6.getInstance( ).talonFXInitialize6(m_shooterLower, "Upper", CTREConfigs6.shooterFXConfig( ));
     m_shooterUpper.setControl(new Follower(m_shooterLower.getDeviceID( ), true));
 
+    SmartDashboard.putNumber("SH_targetRPM", kFlywheelLowerTargetRPM);
+
     initialize( );
   }
 
   @Override
   public void periodic( )
   {
     // This method will be called once per scheduler run
+
+    double current = 0.0;
+
+    // Calculate flywheel RPM and display on dashboard
+    if (m_valid)
+    {
+      m_flywheelRPM = m_flywheelFilter.calculate((m_shooterLower.getVelocity( ).refresh( ).getValue( ) * 60.0));
+
+      m_atDesiredSpeed = (m_flywheelRPM > 30.0) && (Math.abs(m_flywheelTargetRPM - m_flywheelRPM) < kFlywheelToleranceRPM);
+
+      if (m_atDesiredSpeed != m_atDesiredSpeedPrevious)
+      {
+        DataLogManager.log(getSubsystem( ) + ": at desired speed now " + m_flywheelTargetRPM);
+        m_atDesiredSpeedPrevious = m_atDesiredSpeed;
+      }
+
+      current = m_shooterLower.getStatorCurrent( ).refresh( ).getValue( );
+    }
+
+    SmartDashboard.putNumber("SH_flywheelRPM", m_flywheelRPM);
+    SmartDashboard.putBoolean("SH_atDesiredSpeed", m_atDesiredSpeed);
+    SmartDashboard.putNumber("SH_current", current);
+    kFlywheelLowerTargetRPM = SmartDashboard.getNumber("SH_targetRPM", kFlywheelLowerTargetRPM);
```

Comment:
Probably want to do this only when the shooter is set to a new RPM value instead of in periodic all the time.